EventSource/src/util/Processing.ts

```diff
@@ -0,0 +1,123 @@
+    // treePath
+    let mainSubjects: Quad_Subject[] | Set<string> = store.getSubjects(
+        treePath, null, null
+    );
+    const resources = mainSubjects.map(subject => {
+        // extract triples based on subject
+        const resource = store.getQuads(subject, null, null, null) as Resource;
+        // add tree:member
+        resource.push(new Quad(namedNode(eventStreamURI), namedNode(TREE.member), subject));
+        return resource;
+    });
+    mainSubjects = new Set(mainSubjects.map(subj => subj.id));
+    // it's possible for any of resource's object values to be an object further defined here,
+    // if that is the case they get added to this resource
+    for (const quads of resources) {
+        // to avoid issues with data referencing themselves in a circle, duplicates are filtered
+        // out (alongside the current subject)
+        const existingObjects = new Set<string>();
+        existingObjects.add(quads[0].subject.id);
+        for (const quad of quads) {
+            if (existingObjects.has(quad.object.id)) {
+                continue;
+            }
+            existingObjects.add(quad.object.id);
+            // all quads with subjects equal to its object representation gets added to this
+            // resource entry, so the original subjects' data is completely present inside this
+            // single collection of resources
+            // this approach already works recursively, as push adds new elements to the end
+            // quads having another main resource (that is not the current resource) as object
+            // are getting filtered out as well, as they are not further defined
+            quads.push(
+                ...store.getQuads(quad.object, null, null, null).filter((obj) => {
+                    return obj.object.id === quads[0].subject.id || !((mainSubjects as Set<string>).has(obj.object.id))
+                })
+            );
+        }
+    }
+    return resources;
+}
+
+/**
+ * Extracts a single timestamp literal value from the given resource. The treepath is the
+ * predicate associated with the timestamp literal.
+ * 
+ * @param resource Collection of quads where the first subject matches the subject where
+ * the timestamp is requested
+ * @returns {number} The first occurrence of a timestamp literal where predicate matches
+ * the given treepath and the subject matches the very first subject in the quad list
+ * (resource)
+ */
+ export function extractTimeFrom(resource: Resource, treePath: string): number {
+    // as the shape of a resource can vary, a flexible approach using a store is used
+    return extractTimestampFromLiteral(
+        new Store(resource).getObjects(
+            // at this time, all quads in the resource should use the same subject, so
+            // subject of the first quad suffices
+            resource[0].subject, treePath, null
+        )[0] as Literal
+    );
+}
```

Comment:
Isn't this the exact same function as https://github.com/woutslabbinck/SolidEventSourcing/blob/main/EventSource/src/EventSourceUtil.ts#L55-L58

Could you make sure that in this PR only one of those functions remain?

Reply:
This does seem to be the case. I'll remove it and change the references to use that one instead.